### PR TITLE
GEODE-6899: retried client should set last try's version tag if found

### PR DESCRIPTION
**Pull request by gesterzhou**
Co-authored-by: Xiaojian Zhou <gzhou@pivotal.io>
        Co-authored-by: Eric Shu <eshu@pivotal.io>

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/internal/cache/InitialImageOperation.java [hunk 1]
```diff
@@ -1885,10 +1886,8 @@ public boolean executeWith(Object entList, int b) {
           if (lostMemberVersionID == null) {
             lostMemberVersionID = lostMemberID;
           }
-          // check to see if the region in this cache needs to synchronize with others
-          // it is possible that the cache is recover/restart of a member and not
-          // scheduled to synchronize with others
-          synchronizeIfNotScheduled(targetRegion, lostMemberID, lostMemberVersionID);
+          waitInAnotherThreadToCheckIfSynchronizationScheduled(targetRegion, lostMemberID,
```
Comment (pivotal-eshu) [inline geode-core/src/main/java/org/apache/geode/internal/cache/InitialImageOperation.java]: Could check the isRegionSynchronizeScheduledOrDone condition here. If so we do not need to wait.
if (!targetRegion.isRegionSynchronizeScheduledOrDone) {
  waitInAnotherThreadToCheckIfSynchronizationScheduled(targetRegion, lostMemberID, lostMemberVersionID);
}

Comment (gesterzhou) [reply geode-core/src/main/java/org/apache/geode/internal/cache/InitialImageOperation.java]: I would rather let your original code to handle that to reduce code changes.

Comment (bschuchardt) [inline geode-core/src/main/java/org/apache/geode/internal/cache/InitialImageOperation.java]: I don't understand what this has to do with GEODE-6899

Comment (gesterzhou) [reply geode-core/src/main/java/org/apache/geode/internal/cache/InitialImageOperation.java]: This is an enhancement for eshu's previous fix in GEODE-6802. His previous fix is the potential trigger of this bug, i.e. when the newly started member is trigger to call syncWith(). Then make this newly started member contains the entry with version. After that, the retry arrived and create a new version (It should not).

**Pull request merged**